src/main/java/modtrekt/logic/parser/util/CommandParser.java

```diff
@@ -0,0 +1,102 @@
+package modtrekt.logic.parser.util;
+
+import static java.util.Objects.requireNonNull;
+import static modtrekt.commons.util.CollectionUtil.requireAllNonNull;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import modtrekt.commons.util.StringUtil;
+import modtrekt.logic.parser.ArgumentMultimap;
+import modtrekt.logic.parser.ArgumentTokenizer;
+import modtrekt.logic.parser.Prefix;
+import modtrekt.logic.parser.exceptions.ParseException;
+
+/**
+ * CommandParser provides an alternative pathway for parsing commands.
+ * <br>
+ * Unlike ArgumentTokenizer, CommandParser allows validation of required prefixes, as well as
+ * specifying optional prefixes and expected usage message with a fluent API.
+ * Additionally, CommandParser also exposes its utility methods used to validate the arguments of a command.
+ * <br>
+ * The usage message is printed whenever a required prefix is missing via the ParseException that is thrown,
+ * along with a list of the missing prefixes.
+ * <br>
+ * Use CommandParserBuilder to create a CommandParser with a fluent API.
+ */
+public class CommandParser {
```

Comment:
Good job adding this! Think we can integrate this with our other command parsers